service/history/workflow/task_generator.go

```diff
@@ -181,12 +183,18 @@ func (r *TaskGeneratorImpl) GenerateWorkflowCloseTasks(
 	if err != nil {
 		return err
 	}
+	version := currentVersion
+	if deleteAfterClose {
+		// Ignore version check when delete request is from user API
+		version = common.EmptyVersion
```

Comment:
I think this will cause the version check for close execution task to always fail and task being skipped when deleting an open workflow in a global namespace.

Reply:
Good catch. I think it don't need special handling for deleteAfterClose case.

Reply:
The current implementation do check version when delete open workflow(deleteAfterClose = true). Is this expected? @alexshtin